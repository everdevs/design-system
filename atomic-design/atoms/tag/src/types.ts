import { PropsWithTheme } from "@evernest/theme";

export type TagElement = HTMLSpanElement;

export interface StyledTagProps extends PropsWithTheme<TagElement> {}

export interface TagProps extends PropsWithTheme {
	text: string;
}
