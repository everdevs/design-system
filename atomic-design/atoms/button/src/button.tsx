import { css } from "@emotion/core";
import styled from "@emotion/styled";
import { ButtonProps, Size, Variation } from "./types";

const Button = styled("button", {
	shouldForwardProp: (propName: string) => !["theme"].includes(propName),
})<ButtonProps>`
	cursor: pointer;
	border: 0;
	${({ theme, size, variation }) => css`
		box-shadow: ${theme.tokens.shadow.medium.css};
		height: ${size === Size.small ? "20px" : "40px"};
	`};
`;

Button.defaultProps = {
	type: "button",
	size: Size.medium,
	variation: Variation.default,
};

export { Button };
