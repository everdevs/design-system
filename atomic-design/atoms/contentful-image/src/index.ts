export * from "./contentful-image";
export * from "./utils";
export * from "./types";
