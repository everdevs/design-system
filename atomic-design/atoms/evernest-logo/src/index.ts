export * from "./evernest-logo";
export * from "./types";
