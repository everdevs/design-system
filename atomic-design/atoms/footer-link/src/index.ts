export * from "./footer-link";
export * from "./types";
