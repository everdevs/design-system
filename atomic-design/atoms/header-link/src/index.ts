export * from "./header-link";
export * from "./types";
