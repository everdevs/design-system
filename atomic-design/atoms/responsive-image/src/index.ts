export * from "./responsive-image";
export * from "./utils";
export * from "./types";
