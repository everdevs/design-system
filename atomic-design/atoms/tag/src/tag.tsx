import styled from "@emotion/styled";
import React from "react";
import { StyledTagProps, TagElement, TagProps } from "./types";
import { css } from "@emotion/core";

export const StyledTag = styled.span<StyledTagProps>`
	border-radius: 5px;
	padding: var(--spacing-xxxs);
	${({ theme: { palette } }) => css`
		background: ${palette.blue.css};
		color: ${palette.white.css};
	`};
`;

export const Tag = React.forwardRef<TagElement, TagProps>((props, ref) => {
	const {text, ...rest} = props;
	return <StyledTag {...rest} ref={ref}>{text}</StyledTag>
});
