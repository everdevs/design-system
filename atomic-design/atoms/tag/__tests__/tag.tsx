import { createTree, withTheme } from "@evernest/dev-helpers";
import { theme } from "@evernest/theme";
import { mount } from "enzyme";
import React from "react";
import { Tag, TagProps } from "../src";

const ThemedTag = withTheme<TagProps>(Tag, theme);

test("Tag allows setting text", () => {
	const wrapper = mount(<ThemedTag text="New" />);expect(wrapper.text()).toEqual("New");
});

test("Tag matches the snapshot", () => {
	const tree = createTree(<ThemedTag text="foo" />);
	expect(tree).toMatchSnapshot();
});
