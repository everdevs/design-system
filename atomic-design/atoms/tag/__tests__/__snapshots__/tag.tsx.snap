// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag matches the snapshot 1`] = `
<span
  className="css-19e8gv2"
>
  foo
</span>
`;
